File: tp_tda/entregables/5/main.py
def main():
    # Guardo el input
    houses = input()
    city = []
    while houses != 0:
        input_str = input()
        street = [int(i) for i in input_str.split()]
        city.append(street)
        houses = int(input())
    # Termino de guardar el input
    for i in range (len(city)):
        s = 0
        b = 0
        work = 0
        while True: 
            while (s < len(city[i]) and city[i][s] >= 0):
                s+=1 # Mientras no encuentre a alquien que quiera vender avanzo
            while (b < len(city[i]) and city[i][b] <= 0):
                b+=1 # Mientras no encuentre a alguien que quiera comprar avanzo
            if (s == len(city[i]) or b == len(city[i])):
                break # Si llegue al final de las casas corto
            else:
                min_wine = min(city[i][b], -city[i][s]) # Minimo de vinos que se pueden comprar o vender
                work += min_wine * abs(s - b) # Sumo el trabajo que llevo mover los vinos
                city[i][b] -= min_wine # Le resto los vinos que compro
                city[i][s] += min_wine # Le sumo los vinos que vendio
        # Imprimo el resultado
        print(work)

main()

File: tp_tda/entregables/5/main.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);cout.tie(NULL);
    
    long long int houses;
    cin >> houses;
    vector<vector<int>> city;

    while (houses != 0) {
        vector<int> street(houses);
        for (int i = 0; i < houses; ++i) {
            cin >> street[i];
        }
        city.push_back(street);
        cin >> houses;
    }

    for (auto& street : city) {
        long long int s = 0;
        long long int b = 0;
        long long int work = 0;
        while (true) {
            while (s < street.size() && street[s] >= 0) {
                ++s;
            }
            while (b < street.size() && street[b] <= 0) {
                ++b;
            }
            if (s == street.size() || b == street.size()) {
                break;
            } else {
                long long int min_wine = min(street[b], -street[s]);
                work += min_wine * abs(s - b);
                street[b] -= min_wine;
                street[s] += min_wine;
            }
        }
        cout << work << endl;
    }
    return 0;
}
